### PR TITLE
added timeout keyword argument to the http client

### LINKED ISSUE #43: allow configuring timeouts on client requests

### PR DESCRIPTION
**Pull request by z103cb**
Added a timeout kwarg for the http client

Closes #43

### CODE REVIEW

**Comment by dtrifiro** [comment]
Looks good. Can we add a test for it?

File: src/caikit_nlp_client/http_client.py [hunk 1]
```diff
@@ -96,6 +92,7 @@ def generate_text_stream(self, model_id: str, text: str, **kwargs) -> list[str]:
             preserve_input_text (Bool): preserve the input text (default to False)
             max_new_tokens (int): maximum number of new tokens
             min_new_tokens (int): minimum number of new tokens
+            timeout (int): HTTP request timeout value in seconds
```
Comment (dtrifiro) [inline src/caikit_nlp_client/http_client.py:154]:
```python

def generate_text_stream(self, model_id: str, text: str,  timeout: int = 60, **kwargs) -> list[str]:
```

File: src/caikit_nlp_client/http_client.py [hunk 2]
```diff
@@ -110,15 +107,13 @@ def generate_text_stream(self, model_id: str, text: str, **kwargs) -> list[str]:
...
-                self._client_key_path,
-            )
+        requests_kwargs = self._create_request_kwargs(**kwargs)
 
-        response = requests.post(self._api_url, json=json_input, timeout=10.0, **kwargs)
+        response = requests.post(
+            self._stream_api_url,
+            json=json_input,
+            **requests_kwargs,  # nosec
+        )
```
Comment (dtrifiro) [inline src/caikit_nlp_client/http_client.py]:
```suggestion
        response = requests.post(
            self._stream_api_url,
            json=json_input,
            timeout=timeout,
            **requests_kwargs,
        )
```

File: src/caikit_nlp_client/http_client.py (hunk 1 above)
Comment (z103cb) [reply src/caikit_nlp_client/http_client.py:154]: Done

File: src/caikit_nlp_client/http_client.py (hunk 2 above)
Comment (z103cb) [reply src/caikit_nlp_client/http_client.py]: Done!

**Review by dtrifiro — APPROVED** [review]

**Pull request merged**